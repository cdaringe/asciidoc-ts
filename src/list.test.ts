import { describe, expect, test } from "vitest";
import { toAST } from "./mod.js";

test.only("Unordered List", async ({ expect }) => {
  const input = `
* Item 1 **bold**
* Item 2 _italic_
* Item 3 https://foo.org[fooish, aria-role=quz]
  `;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("Ordered List", async ({ expect }) => {
  const input = `
1. First item
2. Second item
  `;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});
