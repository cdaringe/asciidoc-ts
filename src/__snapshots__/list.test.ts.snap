// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Unordered List 1`] = `
{
  "blocks": [
    {
      "items": [
        {
          "content": [
            {
              "content": "Item 1 ",
              "type": "PlainText",
            },
            {
              "content": {
                "content": [
                  {
                    "content": "bold",
                    "type": "PlainText",
                  },
                ],
                "type": "ConstrainedBold",
              },
              "type": "UnconstrainedBold",
            },
          ],
          "type": "UnorderedListItem",
        },
      ],
      "type": "UnorderedList",
    },
    {
      "items": [
        {
          "content": [
            {
              "content": "Item 2 ",
              "type": "PlainText",
            },
            {
              "content": [
                {
                  "content": "italic",
                  "type": "PlainText",
                },
              ],
              "type": "UnconstrainedItalic",
            },
          ],
          "type": "UnorderedListItem",
        },
      ],
      "type": "UnorderedList",
    },
    {
      "items": [
        {
          "content": [
            {
              "content": "Item 3 ",
              "type": "PlainText",
            },
            {
              "attributes": [
                "fooish",
                {
                  "name": "aria-role",
                  "type": "AttributeEntry",
                  "value": "quz",
                },
              ],
              "scheme": "https",
              "type": "UrlMacro",
              "url": "https://foo.org",
            },
          ],
          "type": "UnorderedListItem",
        },
      ],
      "type": "UnorderedList",
    },
  ],
  "type": "Document",
}
`;
